tests/Keboola/DbExtractorConfig/ConfigTest.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\DbExtractorConfig\Tests;
+
+use Keboola\DbExtractorConfig\Config;
+use Keboola\DbExtractorConfig\Configuration\ConfigRowDefinition;
+use Keboola\DbExtractorConfig\Exception\UserException as ConfigUserException;
+use Keboola\DbExtractorConfig\Test\AbstractConfigTest;
+
+class ConfigTest extends AbstractConfigTest
+{
+    public const DRIVER = 'config';
+
+    public function testInvalidConfigTableQuery(): void
+    {
+        $config = $this->getConfig(self::DRIVER);
+        unset($config['parameters']['tables']);
+        $config['parameters']['table'] = [
+            'schema' => 'test',
+            'tableName' => 'test',
+        ];
+        $config['parameters']['query'] = 'select 1 from test';
+        $config['parameters']['outputTable'] = 'fake.output';
+
+        $exceptionMessage = 'Invalid configuration for path "parameters": ';
+        $exceptionMessage .= 'Both table and query cannot be set together.';
+        $this->expectException(ConfigUserException::class);
+        $this->expectExceptionMessage($exceptionMessage);
+
+        $Config = new Config(new ConfigRowDefinition());
+        $Config->validateParameters($config['parameters']);
+    }
+
+    public function testInvalidConfigQueryIncremental(): void
+    {
+        $config = $this->getConfig(self::DRIVER);
+        unset($config['parameters']['tables']);
+        $config['parameters']['incrementalFetchingColumn'] = 'test';
+        $config['parameters']['query'] = 'select 1 from test';
+        $config['parameters']['outputTable'] = 'fake.output';
+
+        $exceptionMessage = 'Invalid configuration for path "parameters": ';
+        $exceptionMessage .= 'Incremental fetching is not supported for advanced queries.';
+        $this->expectException(ConfigUserException::class);
+        $this->expectExceptionMessage($exceptionMessage);
+
+        $Config = new Config(new ConfigRowDefinition());
+        $Config->validateParameters($config['parameters']);
```

Comment:
takova drobna poznamka - expectException a expectExceptionMessage je dobry mit az tesne pred tim radkem, ktery to zpusobuje - protoze takhle muze chyba nastat v `new Config` nebo v `config->validate` a ten test to takhle nerozlisi. 

btw, $Config -> $config